src/librustc_typeck/diagnostics.rs

```diff
@@ -170,6 +171,37 @@ Since `return;` is just like `return ();`, there is a mismatch between the
 function's return type and the value being returned.
 "##,
 
+E0070: r##"
+The left-hand side of an assignment operator must be an lvalue expression. An
+lvalue expression represents a memory location and includes item paths (ie,
+namespaced variables), dereferences, indexing expressions, and field
+references.
+
+More details can be found here:
+https://doc.rust-lang.org/reference.html#lvalues,-rvalues-and-temporaries
+
+Now, we can go further. Here are some bad examples:
+```
+const SOME_CONST : i32 = 12;
+
+fn some_other_func() {}
+
+fn some_function() {
+    SOME_CONST = 14; // error : a constant value cannot be changed !
+    1=3; // error : 1 isn't a valid lvalue !
```

Comment:
Tiny nitpick, but space around the `=` would be nice here.

As much as I love the European separation of the exclamation mark from the word, I think we should follow English convention for consistency. As in: `a constant value cannot be changed!` with the `!` hugging the preceding word.